packages/next/src/utilities/initPage/handleAdminPage.ts

```diff
@@ -48,6 +44,23 @@ export function getRouteInfo({ adminRoute, config, defaultIDType, route }: Args)
       globalConfig = config.globals.find((global) => global.slug === globalSlug)
     }
 
+    // If the collection has an ID field, we need to determine the type of the ID field
+    if (collectionConfig) {
+      const IDField = collectionConfig.fields.find((field) =>
```

Comment:
We do something similar in a ton of different places. Can we access `config.collections[slug].customIDType` from `useConfig` instead? Otherwise this is a lot of looping because we need to call `flattenFields` before the `find` too for situations where somebody has the `id` field in a row or unnamed tab too.

Reply:
Good catch about flattening, I'll converge the `customIDType` logic then